metron-platform/metron-writer/src/main/java/org/apache/metron/writer/bolt/BulkMessageWriterBolt.java

```diff
@@ -74,7 +81,11 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
     this.writerComponent = new BulkWriterComponent<>(collector);
     this.collector = collector;
     super.prepare(stormConf, context, collector);
-    messageGetter = MessageGetters.valueOf(messageGetterStr);
+    if (messageGetStrategyType.equals(MessageGetters.JSON_FROM_FIELD.name()) && messageGetField != null) {
```

Comment:
Why is this special case for JSON_FROM_FIELD required?  Is it because of the arg?  If so, it's probably better to adjust `MessageGetters` rather than doing it here.  I'd suggest making a factory rather than a strategy in this case for it:
```
public enum MessageGetters {

  BYTES_FROM_POSITION(arg -> new BytesFromPosition(arg)),
  JSON_FROM_POSITION(arg -> new JSONFromPosition(arg)),
  JSON_FROM_FIELD(arg -> new JSONFromField(arg));

  Function<Object, MessageGetStrategy> messageGetStrategy;

  MessageGetters(Function<Object, MessageGetStrategy> messageGetStrategy) {
    this.messageGetStrategy = messageGetStrategy;
  }

  public MessageGetStrategy get(Object arg) {
    return messageGetStrategy.apply(arg);
  }
}
```

Then it becomes `MessageGetters.valueOf(messageGetStrategyType).get(messageGetField)`